org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java

```diff
@@ -417,4 +428,36 @@ public boolean setAutoBuilding(boolean enable) throws CoreException {
 		return changed;
 	}
 
+	public List<FileSystemWatcher> registerWatchers() {
+		logInfo(">> registerFeature 'workspace/didChangeWatchedFiles'");
+		if (preferenceManager.getClientPreferences().isWorkspaceChangeWatchedFilesDynamicRegistered()) {
+			Set<String> sources = new HashSet<>();
+			try {
+				IProject[] projects = ResourcesPlugin.getWorkspace().getRoot().getProjects();
+				for (IProject project : projects) {
```

Comment:
we should prolly skip the default project

Reply:
Then, we won't be able to remove '/*.java' and '/src/**' from vscode-java.

Reply:
We'll never be able to remove /*.java anyway, since some folders can contain java files, that are not yet added to the default project. And the default project's source folder is not available to users anyway